copy/copy.go

```diff
@@ -1097,6 +1110,14 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 					Artifact: srcInfo,
 				}
 			}
+
+			// If the reused blob has the same digest as the one we asked for, but
+			// the transport didn't/couldn't supply compression info, fill it in based
+			// on what we know from the srcInfo.
```

Comment:
Please document that this, as a side effect, causes a MIME type update if `LayerInfosForCopy` replaces a blob, because `LayerInfosForCopy`’s `MediaType` update does not cause a manifest edit.

Reply:
Added `If the srcInfos came from LayerInfosForCopy(), then UpdatedImage() will call UpdateLayerInfos(), which uses this information to compute the MediaType value for the updated layer infos, and it the transport didn't pass the information along from its input to its output, then it can derive the MediaType incorrectly.`